pkg/provisioner/controller.go

```diff
@@ -425,14 +409,28 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	} else {
 		ob, err = c.provisioner.Grant(options)
 	}
+	// Record whether the provisioner returned an empty object bucket for error handling use later
+	emptyBucket := (ob == (&v1alpha1.ObjectBucket{}))
```

Comment:
Is it possible that a provisioner returns nil for the ob? I think yes and if so it should be handled, but this can be a separate pr.

Reply:
Sure. I just based this on the previous `else if`

Reply:
Hmm... there is a compilation issue if I do that. It could be that the check as it was written wasn't working as intended. I can also fix this on follow-up since the check was directly copied from elsewhere and not created new for this PR.

```
cannot compare *ob == (v1alpha1.ObjectBucket literal) (operator == not defined for "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1".ObjectBucket)
```

Reply:
Created https://github.com/kube-object-storage/lib-bucket-provisioner/issues/203 for this.